src/sentry/migrations/0779_remove_groups_from_group_inbox.py

```diff
@@ -0,0 +1,54 @@
+# Generated by Django 5.1.1 on 2024-10-24 19:51
+
+from django.apps.registry import Apps
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBarApprox
+
+
+class GroupStatus:
+    RESOLVED = 1
+    IGNORED = 2
+
+
+def remove_groups_from_group_inbox(apps: Apps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    GroupInbox = apps.get_model("sentry", "GroupInbox")
+    Group = apps.get_model("sentry", "Group")
+
+    group_ids_to_delete = []
+    for group_inbox in RangeQuerySetWrapperWithProgressBarApprox(GroupInbox.objects.all()):
+        group = Group.objects.get(id=group_inbox.group_id)
+        if group.status == GroupStatus.RESOLVED or group.status == GroupStatus.IGNORED:
+            group_ids_to_delete.append(group_inbox.group_id)
+
+    GroupInbox.objects.filter(group_id__in=group_ids_to_delete).delete()
```

Comment:
Do you know how many records will need to be cleaned up? If you have more than 10k, I would do deletions in batches to avoid statement timeouts.

Reply:
It's definitely more than 10K - added batching